### PR TITLE
[os-bind] #3650 - break-dnssec toggle needed for Enable filter-aaaa on IPv4/IPv6 clients

### LINKED ISSUE #3650: [os-bind] break-dnssec toggle needed for Enable filter-aaaa on IPv4/IPv6 clients
**Important notices**
Before you add a new report, we ask you kindly to acknowledge the following:

- [x] I have read the contributing guide lines at https://github.com/opnsense/plugins/blob/master/CONTRIBUTING.md
- [x] I have searched the existing issues, open and closed, and I'm convinced that mine is new.
- [x] The title contains the plugin to which this issue belongs

**Describe the bug**
Since there's no way to filter AAAA records with Unbound (FILTER_AAAA is not even compiled in) - this features is often needed for things such as Netflix/HBO/Disney+ with IPv6 tunnels / VPNs), the solution is to forward those queries to local BIND instance listening on another port on localhost with domain overrides in Unbound and set BIND to filter AAAA records. 

However, this does not work when the domain is signed with DNSSEC. For filtering AAAA with DNSSEC-signed domains, the configuration needs to be as follows:

```
plugin query "/usr/local/lib/bind/filter-aaaa.so" {
                   filter-aaaa-on-v4 break-dnssec;
                   filter-aaaa-on-v6 break-dnssec;
           };

```

**To Reproduce**
Enable filter-aaaa on IPv4 Clients/ filter-aaaa on IPv6 Clients
Query AAAA for some DNSSEC-signed domain.

```
; <<>> DiG 9.18.19 <<>> AAAA justice.cz
;; global options: +cmd
;; Got answer:
;; ->>HEADER<<- opcode: QUERY, status: NOERROR, id: 58965
;; flags: qr rd ra; QUERY: 1, ANSWER: 1, AUTHORITY: 0, ADDITIONAL: 1

;; OPT PSEUDOSECTION:
; EDNS: version: 0, flags:; udp: 1232
;; QUESTION SECTION:
;justice.cz.                    IN      AAAA

;; ANSWER SECTION:
justice.cz.             498     IN      AAAA    2001:af0:ffee:200::c2d5:29aa

;; Query time: 4 msec
;; SERVER: 127.0.0.1#53(127.0.0.1) (UDP)
;; WHEN: Sat Nov 04 13:33:57 CET 2023
;; MSG SIZE  rcvd: 67
```

**Expected behavior**
AAAA filtered.

```
# dig AAAA justice.cz

; <<>> DiG 9.18.19 <<>> AAAA justice.cz
;; global options: +cmd
;; Got answer:
;; ->>HEADER<<- opcode: QUERY, status: SERVFAIL, id: 62412
;; flags: qr rd ra; QUERY: 1, ANSWER: 0, AUTHORITY: 0, ADDITIONAL: 1

;; OPT PSEUDOSECTION:
; EDNS: version: 0, flags:; udp: 1232
;; QUESTION SECTION:
;justice.cz.                    IN      AAAA

;; Query time: 30 msec
;; SERVER: 127.0.0.1#53(127.0.0.1) (UDP)
;; WHEN: Sat Nov 04 13:32:15 CET 2023
;; MSG SIZE  rcvd: 39
```

**Additional context**
Probably easiest to implement with DNSSEC Validation set to No in BIND (could be,a separate toggle for the two filter checkboxes as well).

**Environment**
Software version used and hardware type if relevant.
e.g.:

OPNsense 23.7.7_3 (amd64).
os-bind plugin: 1.27_1

### PR DESCRIPTION
**Pull request by doktornotor**
Fixes #3650. This works for me for the mentioned issue and use case. Not exactly familiar with the templates.

### CODE REVIEW

**Review by mimugmail — APPROVED** [review]
Yes, looks good. Thank you!

File: dns/bind/src/opnsense/service/templates/OPNsense/Bind/named.conf [hunk 1]
```diff
@@ -222,10 +222,18 @@ logging {
 {% if helpers.exists('OPNsense.bind.general.filteraaaav4') and OPNsense.bind.general.filteraaaav4 == '1' or helpers.exists('OPNsense.bind.general.filteraaaav6') and OPNsense.bind.general.filteraaaav6 == '1' %}
 plugin query "/usr/local/lib/bind/filter-aaaa.so" {
 {% if helpers.exists('OPNsense.bind.general.filteraaaav4') and OPNsense.bind.general.filteraaaav4 == '1' %}
+{%   if helpers.exists('OPNsense.bind.general.dnssecvalidation') and OPNsense.bind.general.dnssecvalidation == 'no' %}
```
Comment (fichtner) [inline dns/bind/src/opnsense/service/templates/OPNsense/Bind/named.conf:225]: This value is enforced by the model, so the short version is ok:

```suggestion
{%   if OPNsense.bind.general.dnssecvalidation == 'no' %}
```

File: dns/bind/src/opnsense/service/templates/OPNsense/Bind/named.conf [hunk 2]
```diff
@@ -222,10 +222,18 @@ logging {
...
+                   filter-aaaa-on-v4 break-dnssec;
+{%     else %}
                    filter-aaaa-on-v4 yes;
+{%   endif %}
 {% endif %}
 {% if helpers.exists('OPNsense.bind.general.filteraaaav6') and OPNsense.bind.general.filteraaaav6 == '1' %}
+{%   if helpers.exists('OPNsense.bind.general.dnssecvalidation') and OPNsense.bind.general.dnssecvalidation == 'no' %}
```
Comment (fichtner) [inline dns/bind/src/opnsense/service/templates/OPNsense/Bind/named.conf:232]: Here too

**Comment by doktornotor** [comment]
Resolved review and added a missing endif.

File: dns/bind/src/opnsense/service/templates/OPNsense/Bind/named.conf [hunk 3]
```diff
@@ -222,10 +222,18 @@ logging {
 {% if helpers.exists('OPNsense.bind.general.filteraaaav4') and OPNsense.bind.general.filteraaaav4 == '1' or helpers.exists('OPNsense.bind.general.filteraaaav6') and OPNsense.bind.general.filteraaaav6 == '1' %}
 plugin query "/usr/local/lib/bind/filter-aaaa.so" {
 {% if helpers.exists('OPNsense.bind.general.filteraaaav4') and OPNsense.bind.general.filteraaaav4 == '1' %}
+{%   if OPNsense.bind.general.dnssecvalidation == 'no' %}
+                   filter-aaaa-on-v4 break-dnssec;
+{%     else %}
```
Comment (fichtner) [inline dns/bind/src/opnsense/service/templates/OPNsense/Bind/named.conf:227]:
```suggestion
{%   else %}
```

File: dns/bind/src/opnsense/service/templates/OPNsense/Bind/named.conf [hunk 4]
```diff
@@ -222,10 +222,18 @@ logging {
...
+{%   if OPNsense.bind.general.dnssecvalidation == 'no' %}
+                   filter-aaaa-on-v4 break-dnssec;
+{%   else %}
                    filter-aaaa-on-v4 yes;
+{%   endif %}
 {% endif %}
 {% if helpers.exists('OPNsense.bind.general.filteraaaav6') and OPNsense.bind.general.filteraaaav6 == '1' %}
+{%   if OPNsense.bind.general.dnssecvalidation == 'no' %}
+                   filter-aaaa-on-v6 break-dnssec;
+{%     else %}
```
Comment (fichtner) [inline dns/bind/src/opnsense/service/templates/OPNsense/Bind/named.conf:234]:
```suggestion
{%   else %}
```

**Comment by fichtner** [comment]
Merged, thanks!

**Pull request merged**